src/jdbc/cubrid/jdbc/jci/UStatement.java

```diff
@@ -2121,9 +2163,24 @@ public void closeResult() {
 				}
 			}
 			tuples = null;
-			readTupleNumber = 0;
+		}
+	}
+	
+	public void closeResult() {
+		if (stmt_cache_data != null) {
 			getResCache().setExpire();
+			for (int n = 0; n < stmt_cache_data.tuples.size(); n++) {
+				if (tuples == stmt_cache_data.tuples.get(n)) {
```

Comment:
How about testing from the last index of tuples of cached data?
```suggestion
			for (int n = stmt_cache_data.tuples.size(); n > 0; n--) {
				if (tuples == stmt_cache_data.tuples.get(n - 1)) {
```

Reply:
Ok